src/NzbDrone.Core/MediaFiles/MovieImport/Specifications/UpgradeSpecification.cs

```diff
@@ -34,11 +39,46 @@ public Decision IsSatisfiedBy(LocalMovie localMovie, DownloadClientItem download
                 if (qualityCompare < 0)
                 {
                     _logger.Debug("This file isn't a quality upgrade for movie. Skipping {0}", localMovie.Path);
-                    return Decision.Reject("Not an upgrade for existing movie file(s)");
+                    return Decision.Reject("Not a quality upgrade for existing movie file(s)");
+                }
+
+                var customFormats = _customFormatCalculationService.ParseCustomFormat(movieFile);
+                var movieFileCustomFormatScore = localMovie.Movie.Profile.CalculateCustomFormatScore(customFormats);
+
+                if (qualityCompare == 0 && preferredWordScore < movieFileCustomFormatScore)
+                {
+                    _logger.Debug("This file isn't a custom format upgrade for movie. Skipping {0}", localMovie.Path);
+                    return Decision.Reject("Not a custom format upgrade for existing movie file(s)");
                 }
             }
 
             return Decision.Accept();
         }
+
+        private int GetCustomFormatScore(LocalMovie localMovie)
+        {
+            var movie = localMovie.Movie;
+            var scores = new List<int>();
+
+            if (localMovie.FileMovieInfo != null)
+            {
+                var formats = _customFormatCalculationService.ParseCustomFormat(localMovie.FileMovieInfo);
+                scores.Add(movie.Profile.CalculateCustomFormatScore(formats));
+            }
+
+            if (localMovie.FolderMovieInfo != null)
+            {
+                var formats = _customFormatCalculationService.ParseCustomFormat(localMovie.FolderMovieInfo);
+                scores.Add(movie.Profile.CalculateCustomFormatScore(formats));
+            }
+
+            if (localMovie.DownloadClientMovieInfo != null)
+            {
+                var formats = _customFormatCalculationService.ParseCustomFormat(localMovie.DownloadClientMovieInfo);
+                scores.Add(movie.Profile.CalculateCustomFormatScore(formats));
+            }
+
+            return scores.MaxOrDefault();
```

Comment:
I'm wondering if we should union the custom formats and then calculate the score, instead of taking the max of the scores.  It might be that only one of these triggers a format with a negative score.  Not totally sure about this though.

Reply:
Maybe not a bad idea, how does it work when we look at files. Is it filename, scene name, and original unioned?

Reply:
Sorted